### PR TITLE
Remove Creator Onboarding Feature Flags from Codebase

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR removes all instances of `FeatureFlag.enabled?(:creator_onboarding)` from the codebase, specs included, in preparation for the ✨  launch of the Creator Onboarding flow ✨  and refactors necessary code that was once wrapped in the feature flag.

## Related Tickets & Documents
N/A

#### Please note that you will need to essentially clear out your database to take this feature for a spin. 🏎️  

To do so, you can either drop your entire database via `rails:db drop` 

_**or**_

You can open a rails console via `rails c` and do the following:
 ```
# Enable the feature flag
FeatureFlag.enable(:creator_onboarding)

# Delete all users 
User.destroy_all

# Set `waiting_on_first_user` to be true
Settings::General.waiting_on_first_user = true

# If you've logged in before set the mascot id to nil
 Settings::General.mascot_user_id = nil
```

Now, run your server and navigate to `http://localhost:3000/`, click "Create Account," and complete the "Create Account" form. Ensure that you are able to view, complete, and create your account as expected:
Screen Shot 2022-01-06 at 11 38 49 AM

To test the "Email Confirmation" form, please navigate to `http://localhost:3000/confirm-email?email=<emailaddressname>%40<.com>` and ensure that you are able to view and resend the confirmation instructions as expected:
Screen Shot 2022-01-06 at 12 20 39 PM

After completing the "Create Account" form, you should be brought to the new Creator Settings page, `${baseUrl}/admin/creator_settings/new?referrer=${baseUrl}`. Once on the Creator Settings page, ensure that you are able to view, complete, and submit the Creator Settings form:
Screen Shot 2022-01-06 at 11 37 24 AM

Finally, navigate to the `/admin/customization/config` and ensure that you are able to upload a logo (and that it updates in the top, left-hand corner of the site) as expected. Additionally, ensure that the form fields, such as "Primary Brand Color" and "Public" reflect your inputs from the Creator Settings form:
Screen Shot 2022-01-06 at 11 38 03 AM

It wouldn't hurt to click around the app to ensure that nothing else broke along the way, too! 😊 

### UI accessibility concerns?
N/A

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?
**This release will be communicated in the following ways:**
- [x] A changelog post for the community 
- [x] Updated Admin/Creator documentation for the Creator Onboarding flow (a.k.a Initial User Registration) 
- [x] Updated Admin documentation for Images

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

Looney Tunes "That's all folks!"

### CODE REVIEW

File: app/views/admin/settings/forms/_images.html.erb [hunk 1]
```diff
@@ -33,36 +33,7 @@
...
-          <%= admin_config_label :logo_png %>
-          <%= admin_config_description Constants::Settings::General::DETAILS[:logo_png][:description] %>
-          <%= f.text_field :logo_png,
-                           class: "crayons-textfield",
-                           value: Settings::General.logo_png,
-                           placeholder: Constants::Settings::General::DETAILS[:logo_png][:placeholder] %>
-          <% if Settings::General.logo_png.present? %>
-            <img src="<%= Images::Optimizer.call(Settings::General.logo_png, width: 128) %>" width="128" height="128" loading="lazy" />
-          <% end %>
-        </div>
```
Comment (juliannatetreault) [inline app/views/admin/settings/forms/_images.html.erb:46]: Two quick questions:
1. Am I correct in thinking that we want to remove this `logo_png` field (and potentially the related code to the `logo_png` field) when we release the Creator Onboarding flow since we now have the new and improved logo uploader that allows PNGs and JPEGs?
2. If we do want to remove this `logo_png` field, is it fine to do it in this PR or should I open a follow-up PR so that the focus of this PR remains on removing the Creator Onboarding feature flags?
cc @Ridhwana @nickytonline

File: app/views/admin/settings/forms/_images.html.erb [hunk 2]
```diff
@@ -33,36 +33,7 @@
...
-                            value: Settings::General.logo_svg,
-                            rows: 6,
-                            placeholder: Constants::Settings::General::DETAILS[:logo_svg][:placeholder] %>
-            <% if Settings::General.logo_svg.present? %>
-              <div class="site-logo">
-                <%= Settings::General.logo_svg.html_safe %>
-              </div>
-            <% end %>
-          </div>
-        <% end %>
```
Comment (juliannatetreault) [inline app/views/admin/settings/forms/_images.html.erb:65]: Quick question: 
Now that we are no longer excepting `logo_svg`s, is it alright for me to create a follow-up PR that strips the remaining code for `logo_svg`s from the codebase or should I wait until we work on the unified logo RFC that overhauls this anyway?

When I search the codebase, it only appears like there are 9 results, spanning 5 files for `logo_svg`, so a follow-up PR should be relatively quick and straight forward. Thoughts?

cc @Ridhwana @nickytonline

File: app/views/admin/settings/forms/_images.html.erb (hunk 1 above)
Comment (Ridhwana) [reply app/views/admin/settings/forms/_images.html.erb:46]: @juliannatetreault no we don't want to remove `logo_png` at the moment. 

It's completely separate to `logo_svg` is being used in the following places. 

1. The signin modal if you're not logged in but want to comment
2. Some comments view.
3. Onboarding: https://canary1.forem.wtf/onboarding
4. apple-touch-icons
5. sign in invitation when a user is logged out, it shows up on their feed 
6. We expose it via some api calls too.

Perhaps we can do a better job of just explaining the difference with a description for now as it will confuse MANY users. We can tackle this in a separate PR, I'll add it to the board. 

TLDR; 
We will tackle removing logo_png in [this RFC](https://github.com/forem/rfcs/pull/252). At this moment, we will only replace `logo_svg` with `resized_logo`. 

Hope this makes sense :)

File: app/views/admin/settings/forms/_images.html.erb (hunk 2 above)
Comment (Ridhwana) [reply app/views/admin/settings/forms/_images.html.erb:65]: > is it alright for me to create a follow-up PR that strips the remaining code for logo_svgs from the codebase

I think so, yes. But let's merge it only after launch because if something goes wrong with the logos, `logo_svg` can be our fallback.

File: app/views/admin/settings/forms/_images.html.erb (hunk 1 above)
Comment (juliannatetreault) [reply app/views/admin/settings/forms/_images.html.erb:46]: Thank you so much for the clarification, @Ridhwana! I really appreciate the thorough explanation and will revert this change/merge in [your](https://github.com/forem/forem/pull/15989) image PR once it has been shipped. 😊

File: app/views/admin/settings/forms/_images.html.erb (hunk 2 above)
Comment (juliannatetreault) [reply app/views/admin/settings/forms/_images.html.erb:65]: Sounds perfect, @Ridhwana! I'll add an issue to our board so that post-launch, we can tackle this if everything goes well. 😊

**Comment by nickytonline** [comment]
I restarted one of the failed build nodes.

**Comment by juliannatetreault** [comment]
> I restarted one of the failed build nodes.

Thanks! It seems like the failure for the restarted node is actually legitimate @nickytonline, though I'm having trouble running the spec locally--this is the output that I get when I run it:
```
FFFFFF

bundler: failed to load command: rspec (/Users/juliannatetreault/.rbenv/versions/3.0.2/bin/rspec)
/Users/juliannatetreault/.rbenv/versions/3.0.2/lib/ruby/3.0.0/net/http.rb:987:in `initialize': Failed to open TCP connection to 127.0.0.1:9515 (Too many open files - socket(2) for "127.0.0.1" port 9515) (Errno::EMFILE)
```

**Comment by nickytonline** [comment]
> > I restarted one of the failed build nodes.
> 
> Thanks! It seems like the failure for the restarted node is actually legitimate @nickytonline, though I'm having trouble running the spec locally--this is the output that I get when I run it:
> 
> ```
> FFFFFF
> 
> bundler: failed to load command: rspec (/Users/juliannatetreault/.rbenv/versions/3.0.2/bin/rspec)
> /Users/juliannatetreault/.rbenv/versions/3.0.2/lib/ruby/3.0.0/net/http.rb:987:in `initialize': Failed to open TCP connection to 127.0.0.1:9515 (Too many open files - socket(2) for "127.0.0.1" port 9515) (Errno::EMFILE)
> ```

This looks like the error you've received in the past when we've paired. I think this might have been an issue related to your setup?

> Too many open files

**Comment by nickytonline** [comment]
The rspec test is still a valid failure, but I'm still not sure why. When I run the particular failing test, `rspec spec/system/user_uses_the_editor_spec.rb -e "Viewing the editor"`, it says it can't find the text `DEV(local)`, but if I view the source on the page, it's there and in the test itself, if I add `puts page.html`, it's there as well, multiple times in fact. 🤔

```
1) Using the editor Viewing the editor renders the logo or Community name as expected
     Failure/Error: expect(page).to have_text("DEV(local)")
       expected to find text "DEV(local)" in "Create Post\nEdit\nPreview\nClose the editor\nAdd a cover image\nBold CMD + B\nItalic CMD + I\nLink CMD + K\nOrdered list\nUnordered list\nHeading\nQuote\nCode\nCode block\nUpload image\nPublish\nSave draft\nPost options". (However, it was found 11 times including non-visible text.)
     
     [Screenshot Image]: /Users/nickytonline/dev/forem/tmp/screenshots/failures_r_spec_example_groups_using_the_editor_viewing_the_editor_renders_the_logo_or_community_name_as_expected_437.png

     
     # ./spec/system/user_uses_the_editor_spec.rb:29:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:145:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:145:in `block (2 levels) in <top (required)>'
     # ./spec/rails_helper.rb:100:in `block (2 levels) in <top (required)>'

Finished in 13.99 seconds (files took 6.85 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/system/user_uses_the_editor_spec.rb:26 # Using the editor Viewing the editor renders the logo or Community name as expected

[TEST PROF INFO] Time spent in factories: 00:00.451 (3.13% of total time)
Coverage report generated for RSpec to /Users/nickytonline/dev/forem/coverage. 2703 / 23946 LOC (11.29%) covered.
Stopped processing SimpleCov as a previous error not related to SimpleCov has been detected
```

**Comment by juliannatetreault** [comment]
> The rspec test is still a valid failure, but I'm still not sure why. When I run the particular failing test, `rspec spec/system/user_uses_the_editor_spec.rb -e "Viewing the editor"`, it says it can't find the text `DEV(local)`, but if I view the source on the page, it's there and in the test itself, if I add `puts page.html`, it's there as well, multiple times in fact. 🤔

Same! I can see it when I inspect the page, @nickytonline! 🤔

**Comment by nickytonline** [comment]
Alright, I see what the issue is. The `expect(page).to_have_text` matcher is only looking at the document's body's inner text only.

From the dev tools console in the browser.

```
> document.body.innerText
> 'Create Post\nEdit\nPreview\nAdd a cover image\nBold CMD + B\nItalic CMD + I\nLink CMD + K\nOrdered list\nUnordered list\nHeading\nQuote\nCode\nCode block\nUpload image\nWriting a Great Post Title\nThink of your post title as a super short (but compelling!) description — like an overview of the actual post in one short sentence.\nUse keywords where appropriate to help ensure people can find your post by search.\nPublish\nSave draft\nRevert new changes'
```

I think the issue is the site has a logo when the tests run, so the ERB file doesn't hit the else which is the text that would be in the `document.body.innerText`

https://github.com/forem/forem/blob/main/app/views/layouts/_logo.html.erb#L8

```
<a href="<%= root_path %>" class="site-logo" aria-label="<%= t("views.main.aria_home", community: community_name) %>">
  <% if Settings::General.logo_png.present? %>
    <img class="site-logo__img" src="<%= Settings::General.resized_logo %>" alt="<%= community_name %>">
  <% else %>
    <span class="truncate-at-2">
      <%= community_name %>
    </span>
  <% end %>
</a>
```

The test should be updated to check that the logo has the alt text or if there is no logo that the community name is in the page. I'm going to see if we check this in an E2E test already though...

**Comment by nickytonline** [comment]
For another PR, but we cover a lot of what's going on in this test file in end-to-end tests already.

**Comment by juliannatetreault** [comment]
> For another PR, but we cover a lot of what's going on in this test file in end-to-end tests already.

Maybe it's worth removing the failing spec altogether then? I'm trying to work towards a solution now that I was able to get my system specs to run locally again!

**Comment by nickytonline** [comment]
> > For another PR, but we cover a lot of what's going on in this test file in end-to-end tests already.
> 
> 
> 
> Maybe it's worth removing the failing spec altogether then? I'm trying to work towards a solution now that I was able to get my system specs to run locally again!

The spec doesn't add much confidence to using the editor, so I think that particular test could be removed. A follow up PR could be to update one of our E2E tests to ensure the logo is present in general in the app. cc: @Ridhwana

**Comment by juliannatetreault** [comment]
> The spec doesn't add much confidence to using the editor, so I think that particular test could be removed. A follow up PR could be to update one of our E2E tests to ensure the logo is present in general in the app. cc: @Ridhwana

Agreed and sounds good! I'll remove it for now with the plan to add an E2E test that ensures that the logo is present in the general app, like you said, @nickytonline!

File: lib/data_update_scripts/20220105112823_migrate_logo_svg_data.rb [hunk 3]
```diff
@@ -30,9 +30,9 @@ def run
 
       Honeybadger.notify(e, context: {
                            community_name: Settings::Community.community_name,
-                           app_domain: Settings::General.app_domain,
+                           app_domain: Settings::General.app_domain
                          },
-                         tags: "failed_svg_conversion")
+                            tags: "failed_svg_conversion")
```
Comment (juliannatetreault) [inline lib/data_update_scripts/20220105112823_migrate_logo_svg_data.rb:35]: Routine rubocop fixes

**Comment by juliannatetreault** [comment]
_Heads up:_ Once @Ridhwana's `logo_svg` [PR](https://github.com/forem/forem/pull/16103) is merged on Monday, Jan. 17th, I'll have to merge `main` into this PR. 😊

File: app/views/layouts/_logo.html.erb [hunk 4]
```diff
@@ -1,21 +1,18 @@
...
-    <% else %>
-      <span class="site-logo__community-name truncate-at-2">
-        <%= community_name %>
-      </span>
-    <% end %>
+    <span class="site-logo__community-name truncate-at-2">
+      <%= community_name %>
+    </span>
+  <% end %>
+  <% if Settings::General.logo_svg.present? %>
```
Comment (Ridhwana) [inline app/views/layouts/_logo.html.erb:11]: @juliannatetreault lets remove this entire block as well, and then we'll go in and remove all other traces of "logo_svg" in the other issue that you've logged.

**Review by Ridhwana — COMMENTED** [review]
This looks really good @juliannatetreault. I've added some comments based on looking at the code. I haven't tested it as yet on the UI but I'll do that once we address the code changes.

File: app/controllers/async_info_controller.rb [hunk 5]
```diff
@@ -69,7 +69,7 @@ def user_is_a_creator
   end
 
   def use_creator_onboarding
```
[the remark this answers is not in the document]
Comment (juliannatetreault) [reply app/controllers/async_info_controller.rb]: Great catch! I've addressed this.

File: app/views/layouts/_logo.html.erb (hunk 4 above)
Comment (juliannatetreault) [reply app/views/layouts/_logo.html.erb]: Done!

File: app/views/layouts/_logo.html.erb [hunk 6]
```diff
@@ -1,21 +1,19 @@
...
-      it easier to remove code along with the FeatureFlag -->
-  <% if FeatureFlag.enabled?(:creator_onboarding) %>
-    <% if Settings::General.resized_logo.present? %>
-      <img class="site-logo__img" src="<%= Settings::General.resized_logo %>" alt="<%= community_name %>">
-    <% else %>
-      <span class="site-logo__community-name truncate-at-2">
-        <%= community_name %>
-      </span>
-    <% end %>
+  <!-- This is a little explicit at the moment, but it
```
[the remark this answers is not in the document]
Comment (juliannatetreault) [reply app/views/layouts/_logo.html.erb]: ✔️

**Review by Ridhwana — APPROVED** [review]
@juliannatetreault thanks for removing all these feature flags!

I made a couple of other changes and then tested by signing up and logging in with different user. After the change to the registrations_controller above, it seems to work fine. 😊 

I've approved the PR but not sure how much weight that holds now that I've made changes to it as well 😅

**Comment by juliannatetreault** [comment]
_Heads up_: Once this PR is approved, merged, etc. I plan on creating a follow-up PR that removes the `creator_onboarding` feature flag from the codebase via a data update script (if everyone is okay with that!).

**Review by nickytonline — APPROVED** [review]
Thanks for the cleanup @juliannatetreault! ✂️ ✂️ ✂️ ✂️

**Review by citizen428 — APPROVED** [review]

**Review by jgaskins — APPROVED** [review]
Looks good from an SRE perspective! I thought I had one note about the change in the cache key in the application layout (so it would be a guaranteed cold cache in a check that occurs on every single page load) but there are only 2 possible variants of the key, so it'll only be a cache miss 2x/day. False alarm. 😄

**Comment by juliannatetreault** [comment]
> Looks good from an SRE perspective! I thought I had one note about the change in the cache key in the application layout (so it would be a guaranteed cold cache in a check that occurs on every single page load) but there are only 2 possible variants of the key, so it'll only be a cache miss 2x/day. False alarm. 😄

Thanks for the thorough review, @jgaskins! 😄
